### ISSUE TITLE
Failed to generate the Word report: 'NoneType' object is not subscriptable

### ISSUE THREAD

**Issue by kkirsche** [GhostManager/Ghostwriter#60]
```
Failed to generate the Word report: 'NoneType' object is not subscriptable
```

Caused by traceback:

```
ERROR 2020-02-26 17:55:08,714 log 16 140432287571272 Internal Server Error: /reporting/reports/5/generate_docx/
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.7/site-packages/django/core/handlers/base.py", line 115, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/usr/local/lib/python3.7/site-packages/django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python3.7/contextlib.py", line 74, in inner
    return func(*args, **kwds)
  File "/usr/local/lib/python3.7/site-packages/django/contrib/auth/decorators.py", line 21, in _wrapped_view
    return view_func(request, *args, **kwargs)
  File "/app/ghostwriter/reporting/views.py", line 577, in generate_docx
    docx = spenny.generate_word_docx()
  File "/app/ghostwriter/modules/reportwriter.py", line 1284, in generate_word_docx
    self.process_text_xml(finding['description'], finding, report_json)
  File "/app/ghostwriter/modules/reportwriter.py", line 917, in process_text_xml
    self.process_inline_text(line, p, report_type)
  File "/app/ghostwriter/modules/reportwriter.py", line 453, in process_inline_text
    self.span_classes = match[0].split(' ')
TypeError: 'NoneType' object is not subscriptable
```

**Comment by kkirsche** [comment]
Seems to be related to copy and paste situations where TinyMCE retains existing span stylings

**Comment by chrismaddalena** [comment]
Hey, thanks for the contribution! This was a good catch. With everything going on in the world, it's taken me a while to find time a moment to address these open issues and PRs.

The HTML parser was not up to the task to handle all of the potential HTML and XML that could appear in a report. I wrote and tested it against a variety of WYSIWYG generated HTML, but all kinds of unexpected content got thrown at it in production. Your conclusion above is correct; the biggest offenders were blobs of copy/paste text from Word and websites.

This has been addressed in PR #66 with WYSIWYG editor updates and an entirely rewritten HTML parser that no longer relies on string matches. The editor now strips external style information and the new parser uses BeautifulSoup 4 to process everything. I tested this with production data that gave the old parser heaps of trouble and the results were fantastic. The new parser did not encounter issues and consistently handled some of the complex (often bizarre) style combinations the old parser could not reproduce – e.g., a run of inline code that was partially bold/underlined/italicized in different combinations at different points in the run.

I doubt I managed to think of every edge case, but the issues your PRs address should not be issues anymore. I'm going to close the PRs because they can no longer compatible with the latest updates, but know they are super appreciated 😄
